rasa/core/training/story_conflict.py

```diff
@@ -197,8 +197,9 @@ def _find_conflicting_states(
     state_action_mapping = defaultdict(list)
     for element in _sliced_states_iterator(trackers, domain, max_history):
         hashed_state = element.sliced_states_hash
-        if element.event.as_story_string() not in state_action_mapping[hashed_state]:
-            state_action_mapping[hashed_state] += [element.event.as_story_string()]
+        current_hash = hash(element.event)
```

Comment:
if you use `str()` as a substitute for `element.event.as_story_string()` should it be here `str` as well instead of hash?

Reply:
`as_story_string()` is used as hashable, unique event representation to uniquely de-duplicate / identify events. Directly using `hash` makes the intended behavior clearer in my opinion.

Reply:
makes sense, but why do we need `str` then? why not use `hash` everywhere?

Reply:
Sometimes we actually want to give the users a nice presentation of the events (e.g. when telling the users which actions are conflicting).